benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp_ompss/time_step_ompss.cpp

```diff
@@ -0,0 +1,274 @@
+#include <iostream>
+//#include <chrono>
+#include <vector>
+#include <cstdlib>
+
+// Kernels
+#include "../../kernels/c_family/continuity_kern.c"
+#include "../../kernels/c_family/momentum_u_kern.c"
+#include "../../kernels/c_family/momentum_v_kern.c"
+#include "../../kernels/c_family/boundary_conditions_kern.c"
+#include "../../kernels/c_family/time_update_kern.c"
+
+extern "C" void c_invoke_time_step(
+        // Fields
+        double * ssha_t,
+        double * sshn_t,
+        double * sshn_u,
+        double * sshn_v,
+        double * hu,
+        double * hv,
+        double * un,
+        double * vn,
+        double * ua,
+        double * ht,
+        double * ssha_u,
+        double * va,
+        double * ssha_v,
+        // Grid
+        int * tmask,
+        double * area_t,
+        double * area_u,
+        double * area_v,
+        double * dx_u,
+        double * dx_v,
+        double * dx_t,
+        double * dy_u,
+        double * dy_v,
+        double * dy_t,
+        double * gphiu,
+        double * gphiv,
+        // Scalars
+        int istep,
+        int internal_xstart,
+        int internal_xstop,
+        int internal_ystart,
+        int internal_ystop,
+        int width,
+        double rdt,
+        double cbfr,
+        double visc,
+        double omega,
+        double d2r,
+        double g
+        ){
+
+#define TASK_SIZE 32
+//Each double loop over jj then ji is divided into a new structure.
+//The outer loop, now loops over incrememnts of 32 (initially chosen arbitrarliy), and creates tasks of this size
+//For the dependencies supplied to OmpSs, as opposed to putting the full memory for now, we compute the "task row"
+//For the nth task for each jj+=32 loop, this is n.
+//Tasks read or write to the task_rows appropriately (always either task_row, task_row-1, or task_row+1)
```

Comment:
according to their stencil accesses? Presumably they always *write* to `task_row`?